metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py

```diff
@@ -161,11 +161,18 @@ def make_sqlalchemy_uri(
     uri_opts: Optional[Dict[str, Any]] = None,
 ) -> str:
     url = f"{scheme}://"
-    if username is not None:
```

Comment:
we should think about a different way of accomplishing this, adding clickhouse specific code inside sql_common seems like an anti-pattern

Reply:
It looks like the only change is when the `:` separator between the user name and password has to be inserted. This can be accomplished simply by modifying the ClickHouseConfig to override the default value of the password field from the base sql alchemy configuration. I'll update this PR with the change.